ably/rest_client.go

```diff
@@ -360,30 +360,32 @@ func (c *REST) Request(method string, path string, o ...RequestOption) RESTReque
 			default:
 				return nil, fmt.Errorf("invalid HTTP method: %q", method)
 			}
-			var lastResponse *http.Response
 
 			req := &request{
 				Method: method,
 				Path:   path,
 				In:     opts.body,
 				header: opts.headers,
 			}
-			resp, err := c.doWithHandle(ctx, req, func(resp *http.Response, out interface{}) (*http.Response, error) {
-				// Save the resp but return an error on bad status to trigger fallback
-				lastResponse = resp
-				return c.handleResponse(resp, nil)
-			})
-
-			// RSC19e
-			// Only return an error if there was an actual network failiure
-			if err != nil && lastResponse != nil {
-				return lastResponse, nil
-			}
+			resp, err := c.MakeHTTPRequest(ctx, req)
 			return resp, err
 		},
 	}}
 }
 
+func (c *REST) MakeHTTPRequest(ctx context.Context, req *request) (*http.Response, error) {
+	var lastResponse *http.Response
+	resp, err := c.doWithHandle(ctx, req, func(resp *http.Response, out interface{}) (*http.Response, error) {
+		// Save the resp but return an error on bad status to trigger fallback
+		lastResponse = resp
+		return c.handleResponse(resp, nil)
+	})
+	if err != nil && lastResponse != nil {
+		return lastResponse, nil
+	}
+	return resp, err
```

Comment:
I looked for other uses of `REST.doWithHandle`, and there is [`REST.do`](https://github.com/ably/ably-go/blob/main/ably/rest_client.go#L599-L601) which I think does exactly what this method is doing (i.e. sending a request, and passing the response to `REST.handleResponse`).

Also, it looks like `REST.handleResponse` will call `decodeResp` if `req.Out` is set, so that should avoid the need to do it in `RESTChannel.Status`.

Also, this is why I'm encouraging a refactor here, there is just way too much indirection here!